Sources/SwiftDocC/Infrastructure/Link Resolution/PathHierarchy.swift

```diff
@@ -148,7 +148,48 @@ struct PathHierarchy {
                 // would require that we redundantly create multiple nodes for the same symbol in many common cases and then merge them. To avoid doing that, we instead check
                 // the source symbol's path components to find the correct target symbol by matching its name.
                 if let targetNode = nodes[relationship.target], targetNode.name == expectedContainerName {
-                    targetNode.add(symbolChild: sourceNode)
+                    if sourceNode.parent == nil || sourceNode.parent === targetNode {
```

Comment:
We could do nothing if the target already is the parent of the source, otherwise we might hit the added assertion.